Modules/_threadmodule.c

```diff
@@ -1110,8 +1110,12 @@ Release the lock.");
 static PyObject *
 rlock_locked(PyObject *op, PyObject *Py_UNUSED(ignored))
 {
-    rlockobject *self = rlockobject_CAST(op);
-    int is_locked = _PyRecursiveMutex_IsLockedByCurrentThread(&self->lock);
+    /*
+    see gh-134323: the `_PyRecursiveMutex_IsLocked` function does not exist, so we cast the `op`
+    to `lockobject` in order to call `PyMutex_IsLocked`.
+    */
+    lockobject *self = lockobject_CAST(op);
+    int is_locked = PyMutex_IsLocked(&self->lock);
```

Comment:
This works because the first field of `_PyRecursiveMutex` is a `PyMutex`, but it's a bit confusing. I'd write this as:

```c
rlockobject *self = rlockobject_CAST(op);
int is_locked = PyMutex_IsLocked(&self->lock.mutex);
```